### PR TITLE
Feature/r1cs import

### PR DESCRIPTION
**Pull request by JacobEberhardt**
Import of Standard R1CS files for interoperability.

### CODE REVIEW

File: src/r1cs.rs [hunk 1]
```diff
@@ -331,6 +344,80 @@ pub fn r1cs_program<T: Field>(
...
+/// * The Rank 1 Constraint System (R1CS) is defined as:
+/// * `<A,x>*<B,x> = <C,x>` for a witness `x`
+/// * Since the matrices in R1CS are usually sparse, the following encoding is used:
+/// * For each constraint (i.e., row in the R1CS), only non-zero values are supplied and encoded as a tuple (index, value).
+///
+/// # Arguments
+///
+/// * r1cs - R1CS in standard JSON data format
+
+pub fn flattened_program<T: Field>(
```
Comment (Schaeff) [inline src/r1cs.rs]: Could this be in an `impl` for `R1CS` to call `R1CS::from(json).to_flattened_program()` ?

Comment (JacobEberhardt) [reply src/r1cs.rs]: Yes, good idea. Let's add that after the format has been standardized and the data structure fixed.

Comment (Schaeff) [reply src/r1cs.rs]: You mean not in this PR?

Comment (JacobEberhardt) [reply src/r1cs.rs]: That depends if we want to merge this preliminary version or finalize it first.

**Comment by nicola** [comment]
woho! this is big!

**Comment by Kyroy** [comment]
jenkins-test

**Pull request merged**